libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/CaseInsensitiveComparison.java

```diff
@@ -16,18 +16,10 @@ class CaseInsensitiveComparison extends FileAccessTreeComparison {
     }
 
     /**
-     * Case-insensitive character comparison. Taken from the JDK {@code WindowsPath#compareTo} implementation.
+     * Case-insensitive character comparison. See also {@code WindowsPath#compareTo}.
      */
     private static int caseInsensitiveCharacterComparator(char c1, char c2) {
-        if (c1 == c2) {
-            return 0;
-        }
-        c1 = Character.toUpperCase(c1);
-        c2 = Character.toUpperCase(c2);
-        if (c1 == c2) {
-            return 0;
-        }
-        return Character.compare(c1, c2);
+        return Character.compare(Character.toLowerCase(c1), Character.toLowerCase(c2));
```

Comment:
Is there a significance to the fact that we switched from uppercase to lowercase?

Reply:
Nope, just a suggestion from Ryan to make it shorter and more readable

Reply:
`toUpperCase` is equally short and readable, no?

I just think if we use one that's different from the actual filesystem code you copied, we're asking for unexpected corner cases.